tezos_interop/tezos_interop.re

```diff
@@ -131,6 +131,15 @@ module Key = {
     // TODO: move this to a functor
     obj1(req(name, raw_encoding));
   };
+  let to_cstruct =
+    fun
+    | Ed25519(pub_) => Mirage_crypto_ec.Ed25519.pub_to_cstruct(pub_);
+  let equal = (a, b) => {
+    switch (a, b) {
+    | (Ed25519(_) as a, Ed25519(_) as b) =>
```

Comment:
I think the keys have to be unwrapped from the respective Tags/Variants and then converted to cstruct right away. Instead of passing down the wrapped key to `to_cstruct`. This will make it easier in future when we support more curves

Reply:
I also prefer the simple one, I know it leads to slightly more duplicated code, but I also wonder, why this is not generated by `[@deriving eq]`?

Reply:
added explicit equal fn to Ed25519 module for public keys and secret keys and used ```@deriving eq``` here